classes/class-pmpro-field.php

```diff
@@ -747,8 +756,26 @@ function getHTML($value = "")
 		//anything meant to be added to the beginning or end?
 		$r = $r_beginning . $r . $r_end;
 
-		//filter
-		$r = apply_filters('pmprorh_get_html', $r, $this);
+		/**
+		 * Allow hooking into the generated field HTML for a specific field type.
+		 *
+		 * @since TBD
+		 *
+		 * @param string      $r     The field HTML.
+		 * @param mixed       $value The field value.
+		 * @param PMPro_Field $field The field object.
+		 */
+		$r = apply_filters( "pmpro_field_get_html_{$this->type}", $r, $value, $this );
+
+		/**
+		 * Allow hooking into the generated field HTML.
+		 *
+		 * @since TBD
+		 *
+		 * @param string      $r     The field HTML.
+		 * @param PMPro_Field $field The field object.
+		 */
+		$r = apply_filters( 'pmprorh_get_html', $r, $this );
```

Comment:
```suggestion
		$r = apply_filters( 'pmpro_field_get_html', $r, $this );
```
We can also add a deprecated filter that maps to this.